executor/builder.go

```diff
@@ -657,11 +657,11 @@ func (b *executorBuilder) buildSelectLock(v *plannercore.PhysicalLock) Executor
 		defer func() { b.inSelectLockStmt = false }()
 	}
 	b.hasLock = true
-	if b.err = b.updateForUpdateTSIfNeeded(v.Children()[0]); b.err != nil {
+
+	// Build 'select for update' using the 'for update' ts.
+	if b.forUpdateTS, b.err = b.getSnapshotTS(); b.err != nil {
```

Comment:
Why do you change "updateForUpdateTS" to "getSnapshotTS()"? 
Does it update the latest TS for "SELECT ... FOR UPDATE" in original code, but not now.

Reply:
`getSnapshotTS` can also update for_update_ts. The naming is somewhat confusing..